CMakeLists.txt

```diff
@@ -354,13 +354,58 @@ if(GINKGO_BUILD_BENCHMARKS)
 endif()
 
 if(GINKGO_DEVEL_TOOLS)
-    add_custom_target(add_license
-        COMMAND ${Ginkgo_SOURCE_DIR}/dev_tools/scripts/add_license.sh
-        WORKING_DIRECTORY ${Ginkgo_SOURCE_DIR})
-    # if git-cmake-format can not build format target, do not add the dependencies
-    if(TARGET format)
+    function(enable_format_check)
+        macro(message_git_warning)
+            message(WARNING "No viable Git setup found, but GINKGO_DEVEL_TOOLS=ON is set. "
+                    "It is recommended to enable this option only with the intention of "
+                    "adding changes to Ginkgo via Git.")
```

Comment:
I think we should fail here

Reply:
I'm open to that, and that would make the code quite a bit easier (I wouldn't need the function anymore). But I think we don't fail currently in this case, so I kept that behavior.